firebase-firestore/src/test/java/com/google/firebase/firestore/FilterTest.java

```diff
@@ -0,0 +1,188 @@
+    assertNotEquals(filter, Filter.greaterThanOrEqualTo("y", "v"));
+    assertNotEquals(filter,Filter.lessThanOrEqualTo("y", "v"));
+  }
+
+  @Test
+  public void lessThan() {
+    Filter filter = Filter.lessThan("x", "v");
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.lessThan(FieldPath.of("x"), "v"));
+    assertNotEquals(filter, Filter.lessThan("x", "z"));
+    assertNotEquals(filter, Filter.lessThan("y", "v"));
+    assertNotEquals(filter,Filter.greaterThan("y", "v"));
+  }
+
+  @Test
+  public void lessThanOrEqualTo() {
+    Filter filter = Filter.lessThanOrEqualTo("x", "v");
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.lessThanOrEqualTo(FieldPath.of("x"), "v"));
+    assertNotEquals(filter, Filter.lessThanOrEqualTo("x", "z"));
+    assertNotEquals(filter, Filter.lessThanOrEqualTo("y", "v"));
+    assertNotEquals(filter,Filter.greaterThanOrEqualTo("y", "v"));
+  }
+
+  @Test
+  public void arrayContains() {
+    Filter filter = Filter.arrayContains("x", "v");
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.arrayContains(FieldPath.of("x"), "v"));
+    assertNotEquals(filter, Filter.arrayContains("x", "z"));
+    assertNotEquals(filter, Filter.arrayContains("y", "v"));
+    assertNotEquals(filter,Filter.equalTo("y", "v"));
+  }
+
+  @Test
+  public void arrayContainsAny() {
+    Filter filter = Filter.arrayContainsAny("x", ImmutableList.of("v1", "v2"));
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.arrayContainsAny(FieldPath.of("x"), ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.arrayContainsAny("x", ImmutableList.of("v2", "v3")));
+    assertNotEquals(filter, Filter.arrayContainsAny("y", ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.equalTo("y", "v"));
+  }
+
+  @Test
+  public void inArray() {
+    Filter filter = Filter.inArray("x", ImmutableList.of("v1", "v2"));
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.inArray(FieldPath.of("x"), ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.inArray("x", ImmutableList.of("v2", "v3")));
+    assertNotEquals(filter, Filter.inArray("y", ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.notInArray("y", ImmutableList.of("v1", "v2")));
+  }
+
+  @Test
+  public void notInArray() {
+    Filter filter = Filter.notInArray("x", ImmutableList.of("v1", "v2"));
+    assertEquals(filter, filter);
+    assertEquals(filter, Filter.notInArray(FieldPath.of("x"), ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.notInArray("x", ImmutableList.of("v2", "v3")));
+    assertNotEquals(filter, Filter.notInArray("y", ImmutableList.of("v1", "v2")));
+    assertNotEquals(filter, Filter.inArray("y", ImmutableList.of("v1", "v2")));
+  }
+
+  @Test
+  public void or() {
+    Filter filter = Filter.or(
+        Filter.equalTo("x", ImmutableList.of("v1", "v2")),
+        Filter.equalTo("y", ImmutableList.of("v3", "v4"))
```

Comment:
technically, it's not correct to pass a List as a value of the `equalTo` filter operator. I believe it'd error out when the filter gets parsed. It doesn't really make a different in the context of what is being unit tested here though. But to be precise, let's replace `equalTo` with `inArray`.

Applies elsewhere below.

Reply:
Will do.